boot-core/tests/contract.rs

```diff
@@ -0,0 +1,223 @@
+/*
+
+*/
+#[cfg(test)]
+mod contract {
+    use std::{
+        env,
+        time::Duration,
+        sync::Mutex,
+        sync::Arc,
+        thread::sleep,
+    };
+
+    use ctor::{ctor, dtor};
+    use once_cell::sync::Lazy;
+    use duct::cmd;
+
+    use tokio::runtime::Runtime;
+
+    use cosmwasm_std::Uint128;
+    use cw_multi_test::ContractWrapper;
+
+    use speculoos::prelude::*;
+
+    use boot_core::{
+        contract,
+        networks::LOCAL_JUNO,
+        Contract, DaemonOptionsBuilder,
+        instantiate_daemon_env,
+    };
+
+    use cw20_base::msg::*;
+
+    static mut DOCKER_CONTAINER_ID: Lazy<String> = Lazy::new(|| String::new());
+    static INIT: Lazy<Mutex<bool>> = Lazy::new(|| Mutex::new(false));
+
+    // const CW20_CONTRACT_WASM: &str = "/../boot-cw-plus/cw-artifacts/cw20_base.wasm";
+    const CW20_CONTRACT_WASM: &str = "/../boot-cw-plus/cw-artifacts/cw20_base.wasm";
+    const STATE_FILE: &str = "/tmp/boot_test.json";
+    const LOCAL_MNEMONIC: &str = "clip hire initial neck maid actor venue client foam budget lock catalog sweet steak waste crater broccoli pipe steak sister coyote moment obvious choose";
+
+    #[ctor]
```

Comment:
This code should remain in the `common/mod.rs` file as it can serve as the basis for all our integration tests.